### PR TITLE
Feature: Maven Archetype and custom plugin for new modules

### PR DESCRIPTION
**Pull request by 0-Tikaro**
- [ ] Do the individual modules require readme files?

### CODE REVIEW

File: s-pipes-modules-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <relativePath>../s-pipes-parent</relativePath>
+        <groupId>cz.cvut.kbss</groupId>
+        <artifactId>s-pipes-parent</artifactId>
+        <version>0.3.0</version>
```
Comment (blcham) [inline s-pipes-modules-utils/pom.xml:9]:
```suggestion
        <version>0.4.0</version>
```

File: s-pipes-modules-utils/regen.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+BASEDIR="/Users/qwerty/Documents/documents/university/masters/semester-4/semestral-project/personal-fork/s-pipes-modulegen/s-pipes-modules-utils/"
```
Comment (blcham) [inline s-pipes-modules-utils/regen.sh:2]:
```
BASEDIR=$(dirname $(readlink -m $0))
SPIPESDIR=$BASEDIR/../s-pipes-modules
```

File: s-pipes-modules-utils/s-pipes-module-archetype/pom.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>cz.cvut.kbss</groupId>
+        <artifactId>s-pipes-modules-utils</artifactId>
+        <version>0.3.0</version>
```
Comment (blcham) [inline s-pipes-modules-utils/s-pipes-module-archetype/pom.xml:8]:
```suggestion
        <version>0.4.0</version>
```

**Review by blcham — COMMENTED** [review]
Not done yet.

File: s-pipes-modules-utils/s-pipes-module-archetype/src/main/resources/archetype-resources/src/test/java/__moduleNameCamelCase__ModuleTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    private $className module;
+
+    @BeforeEach
+    public void setUp() {
+        module = new $className();
+    }
+
+    @Test
+    public void loadConfiguration_doesNotThrow() {
```
Comment (blcham) [inline s-pipes-modules-utils/s-pipes-module-archetype/src/main/resources/archetype-resources/src/test/java/__moduleNameCamelCase__ModuleTest.java:25]:
```suggestion
    void loadConfiguration_doesNotThrow() {
```

Comment (blcham) [reply s-pipes-modules-utils/s-pipes-module-archetype/src/main/resources/archetype-resources/src/test/java/__moduleNameCamelCase__ModuleTest.java:25]: To be more concise. Do not need to be public anymore from some version of junit lib.

File: s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/pom.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>cz.cvut.kbss</groupId>
+        <artifactId>s-pipes-modules-utils</artifactId>
+        <version>0.3.0</version>
```
Comment (blcham) [inline s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/pom.xml:8]:
```suggestion
        <version>0.4.0</version>
```

File: s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    </parent>
+
+    <name>SPipes Module Annotation Processor</name>
+    <artifactId>s-pipes-module-creator-maven-plugin</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (blcham) [inline s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/pom.xml:13]:
```suggestion
    <version>0.4.0</version>
```

File: s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/src/main/java/RdfAnnotationProcessorMojo.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                model.add(modelConstraint, RDF.type, SPL.Argument);
+                model.add(modelConstraint, SPL.predicate, annotation.urlPrefix() + annotation.name());
+                model.add(modelConstraint, RDFS.comment, "Automatically generated field: " + annotation.name());
+                model.add(subject, SPIN.constraint, modelConstraint);
+
+                log.info("Added model constraint based on annotation: " +
+                        "(name = " + annotation.name() + ", urlPrefix = " + annotation.urlPrefix() + ")");
+            }
+        }
+        model.write(new FileOutputStream(ontologyFilepath), "TTL");
```
Comment (blcham) [inline s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/src/main/java/RdfAnnotationProcessorMojo.java:96]: "TTL" --> FileUtils.langTurtle

File: s-pipes-modules-utils/s-pipes-module-archetype/src/main/resources/archetype-resources/pom.xml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>cz.cvut.kbss</groupId>
+                <artifactId>s-pipes-module-creator-maven-plugin</artifactId>
+                <version>1.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (0-Tikaro) [reply s-pipes-modules-utils/s-pipes-module-archetype/src/main/resources/archetype-resources/pom.xml:45]: Yeah, this makes sense - it also means that these new modules have the same versioning pattern as the other modules in the repo.

File: s-pipes-modules-utils/regen.sh [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (0-Tikaro) [reply s-pipes-modules-utils/regen.sh]: The `s-pipes-module-utils` is a wrapper to hold the archetype and the plugin together without them just floating in the root of the project. The script was actually just something I used for testing, there was indeed no reason for it to be in that folder. Moved it to `s-pipes-module-archetype/bin` and reworked it a bit.

**Review by blcham — APPROVED** [review]

**Pull request merged**